sdk/textanalytics/ai-text-analytics/src/paging.ts

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { PagedAsyncIterableIterator, PageSettings } from "@azure/core-paging";
+
+/**
+ * an interface that describes how to communicate with the service and how to build a page of items.
+ */
+export interface PagedResult<TOptions extends Record<string, any>, TResponse, TPage> {
```

Comment:
nit
```suggestion
export interface PagedResult<TOptions extends Record<string, unknown>, TResponse, TPage> {
```

Reply:
This change will cause the following error in client code: `Type 'OperationOptions' is not assignable to type 'Record<string, unknown>'`

Reply:
Ugh, this apparently is a known issue when attempting to use `Record` as a generic object shape while passing in an interface: https://github.com/Microsoft/TypeScript/issues/15300